plugins/deployment_manager/Deploy.ts

```diff
@@ -21,6 +22,7 @@ export interface DeployOpts {
   overwrite?: boolean; // should we overwrite existing contract link
   connect?: Signer; // signer for the returned contract
   verify?: boolean; // verify contract on etherscan
+  lazyVerify?: boolean; // delay verification: can prevent flakiness of deployments
```

Comment:
I'm thinking of consolidating these two options into one with an enum value instead. Feels kind of weird that you can specify both `verify` and `lazyVerify` right now. Instead, I think it's a nicer interface if you specify either `VerifyOption.EAGER`, `VerifyOption.LAZY`, `VerifyOption.NONE`.

Reply:
Agree the 2 options are part of awkwardness and the states you can get into with it are weird now

Reply:
Combined these options into a union type `VerificationStrategy`.